packages/backend-app-api/src/config/config.ts

```diff
@@ -109,12 +111,12 @@ export async function loadBackendConfig(options: {
       }),
     },
   });
-
   console.info(
     `Loaded config from ${appConfigs.map(c => c.context).join(', ')}`,
   );
 
   config.setConfig(ConfigReader.fromConfigs(appConfigs));
+  config.mergeConfig(options.config ?? {});
```

Comment:
Rather than operating on the completed config reader, I think it's better if we have this happen at the `AppConfig` level instead on the line above. The `AppConfig` represents raw config data along with some context around how it was loaded.